File: static/js/systemInfo.js
function bytesToSize(bytes) {
    const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB'];
    if (bytes === 0) return '0 Bytes';
    const i = parseInt(Math.floor(Math.log(bytes) / Math.log(1024)), 10);
    return Math.round(bytes / Math.pow(1024, i), 2) + ' ' + sizes[i];
}

async function fetchSystemInfo() {
    try {
        const response = await fetch('/system_info');
        const data = await response.json();

        setProgressBar('#cpu-progress-bar', data.cpu_percent);
        setProgressBar('#memory-progress-bar', data.memory_percent);
        setProgressBar('#disk-progress-bar', data.disk_percent);

        document.getElementById('java-version').innerText = `Java Version: ${data.java_version}`;
        document.getElementById('cpu-percentage').innerText = `CPU: ${data.cpu_percent}%`;
        document.getElementById('memory-percentage').innerText = `Memory: ${data.memory_percent}% (${bytesToSize(data.current_ram_used)} used / ${bytesToSize(data.total_ram)} total)`;
        document.getElementById('disk-percentage').innerText = `Disk: ${data.disk_percent}% (${bytesToSize(data.current_disk_used)} used / ${bytesToSize(data.total_disk)} total)`;

    } catch (error) {
        console.error('Error fetching system info:', error);
    }
}

function setProgressBar(selector, percent) {
    const progressBarFill = document.querySelector(selector);
    progressBarFill.style.width = `${percent}%`;

    if (percent > 80) {
        progressBarFill.style.backgroundColor = 'red';
    } else if (percent > 60) {
        progressBarFill.style.backgroundColor = 'orange';
    } else {
        progressBarFill.style.backgroundColor = 'green';
    }
}

// Call the fetchSystemInfo function every minutes
setInterval(fetchSystemInfo, 3800);

// Call the fetchSystemInfo function once on page load
fetchSystemInfo();

File: routes/systemInfo.py
import re
import subprocess
import psutil
from flask import jsonify

def init_system_info_routes(app):
    @app.route('/system_info')
    def system_info():
        cpu_percent = psutil.cpu_percent()
        memory_percent = psutil.virtual_memory().percent
        disk_percent = psutil.disk_usage('/').percent

        try:
            try:
                java_version_output = subprocess.check_output(['java', '--version'], stderr=subprocess.STDOUT)
                java_version_match = re.search(r'openjdk (\d+\.\d+\.\d+)', java_version_output.decode('utf-8'))
            except:
                java_version_output = subprocess.check_output(['java', '-version'], stderr=subprocess.STDOUT)
                java_version_match = re.search(r'openjdk version "(\d+\.\d+\.\d+_\d+)"', java_version_output.decode('utf-8'))
            java_version = java_version_match.group(1) if java_version_match else "Unknown"
        except Exception as e:
            java_version = str(e)

        total_ram = psutil.virtual_memory().total
        total_disk = psutil.disk_usage('/').total
        current_ram_used = psutil.virtual_memory().used
        current_disk_used = psutil.disk_usage('/').used

        return jsonify({
            'cpu_percent': cpu_percent,
            'memory_percent': memory_percent,
            'disk_percent': disk_percent,
            'java_version': java_version,
            'total_ram': total_ram,
            'total_disk': total_disk,
            'current_ram_used': current_ram_used,
            'current_disk_used': current_disk_used
        })